source/common/router/upstream_request.cc

```diff
@@ -111,14 +111,24 @@ UpstreamRequest::UpstreamRequest(RouterFilterInterface& parent,
 
   // The router checks that the connection pool is non-null before creating the upstream request.
   auto upstream_host = conn_pool_->host();
+  absl::optional<Upstream::ClusterInfoConstSharedPtr> cluster_info =
+      parent_.callbacks()->streamInfo().upstreamClusterInfo();
+
   Tracing::HttpTraceContext trace_context(*parent_.downstreamHeaders());
+  Tracing::UpstreamContext upstream_context(upstream_host, // host_
+                                            cluster_info.has_value() ? cluster_info.value()
+                                                                     : nullptr, // cluster_info_
+                                            Tracing::ServiceType::Unknown,      // service_type_
+                                            false                               // is_side_stream_
```

Comment:
This still won't work as this path is shared between sidestream and upstream. 

If you really need differentiation here to work, you will need extra info populated to `UpstreamRequest`

Reply:
Good point, thanks. I realized yesterday, taking a further look.

Reply:
Thanks for review! I think the key point here it is fine to only distinguish the callside from async client, change the name and comment.

Reply:
The key problem is that async client can be the caller here as well. So I think changing the name/comment won't make the difference as the underlying issue is still same.

sidestream (e.g., envoy-gRPC ) is on top of the async client, that is what I meant this path is shared with sidestream and upstream. Perhaps there are some misunderstandings?

Please see my the other comment.